services/src/api/model/db_types.rs

```diff
@@ -364,6 +363,40 @@ impl TryFrom<SymbologyDbType> for Symbology {
     }
 }
 
+#[derive(Debug, PartialEq, ToSql, FromSql)]
+pub struct TextTextKeyValue {
+    key: String,
+    value: String,
+}
+
+#[derive(Debug, PartialEq, ToSql, FromSql)]
+#[postgres(transparent)]
+pub struct HashMapTextTextDbType(pub Vec<TextTextKeyValue>);
+
+impl From<&HashMap<String, String>> for HashMapTextTextDbType {
+    fn from(map: &HashMap<String, String>) -> Self {
+        Self(
+            map.iter()
+                .map(|(key, value)| TextTextKeyValue {
+                    key: key.clone(),
+                    value: value.clone(),
+                })
+                .collect(),
+        )
+    }
+}
+
+impl<S: std::hash::BuildHasher + std::default::Default> From<HashMapTextTextDbType>
+    for HashMap<String, String, S>
+{
+    fn from(map: HashMapTextTextDbType) -> Self {
+        map.0
+            .into_iter()
+            .map(|TextTextKeyValue { key, value }| (key, value))
```

Comment:
what happens if a key occurs twice? There is no constraint that prevents this in postgres and it shouldn't crash our program. 

Do we need a test that checks this?

Reply:
First it does not crash: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d9616410a6c778fb727d182d6799e8f7

Second, since we only input hashmaps, I wouldn't check it for now in Postgres.